db/_integration/db_test.go

```diff
@@ -0,0 +1,81 @@
+package integration
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/packethost/pkg/log"
+	"github.com/pkg/errors"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"github.com/tinkerbell/tink/db"
+)
+
+type NewPostgresDatabaseRequest struct {
+	ApplyMigration bool
+}
+
+// NewPostgresDatabaseClient returns a SQL client ready to be used. Behind the
+// scene it is starting a Docker container that will get cleaned up when the
+// test is over. Tests using this function are safe to run in parallel
+func NewPostgresDatabaseClient(t *testing.T, ctx context.Context, req NewPostgresDatabaseRequest) (*sql.DB, *db.TinkDB, func() error) {
+	postgresC, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: testcontainers.ContainerRequest{
+			Image:        "postgres:13.1",
+			ExposedPorts: []string{"5432/tcp"},
+			WaitingFor:   wait.ForLog("database system is ready to accept connections"),
```

Comment:
we should add support to wait for something like wait.Exec and then we can just call `pg_isready`.